src/operator/nn/mkldnn/mkldnn_pooling.cc

```diff
@@ -224,7 +226,7 @@ MKLDNNPoolingFwd& GetPoolingFwd(const PoolingParam& param,
       pooling_fwds;
 #endif
 
-  bool with_workspace = is_train && MKLDNNRequireWorkspace(param);
+  const bool with_workspace = is_train && (MKLDNNRequireWorkspace(param) || use_adaptive_pooling);
```

Comment:
Could you please tell me why you placed `use_adaptive_pooling` here?

Reply:
Because this logic is merged from the adaptive pooling and condition there was:
`bool with_workspace = is_train;`